src/pages/404.tsx

```diff
@@ -1,43 +1,47 @@
-import { graphql } from 'gatsby';
+import { StaticQuery, graphql } from 'gatsby';
 import React from 'react';
 import Hero from '../components/hero';
 import Layout from '../components/layout';
 import Navigation from '../components/navigation';
 import { LearnPageData } from '../types';
 import { findActive } from '../util/findActive';
 
-type Props = {
-  data: LearnPageData;
-};
-
-export default ({ data }: Props) => {
+export default () => {
   // Get current page related data.
-  const { navigationSections } = findActive(data.sections.group);
-
   const title = `PAGE NOT FOUND`;
   const description = `You've hit a route that does not exist.`;
 
   return (
-    <Layout title={title} description={description}>
-      <Hero title={title} />
-      <Navigation sections={navigationSections} />
-      <article className="article-reader">
-        <h1 className="article-reader__headline">{title}</h1>
-        <div>
-          <p>
-            The page you're trying to access does not exist. Go back to the
-            Homepage or find what you're looking for in the menu.
-          </p>
-          <p>
-            Take me back to the <a href="/">Homepage</a> →
-          </p>
-        </div>
-      </article>
-    </Layout>
+    <StaticQuery
+      query={query}
+      render={(data: LearnPageData) => {
```

Comment:
nit: Can we extract out the render function into a `const`? Makes `<StaticQuery>` simpler.

```ts
<StaticQuery
  query={query}
  render={<NotFound />}
/>

const NotFound = (data: LearnPageData) => {
  const { navigationSections } = findActive(data.sections.group);

  return (
    <Layout title={title} description={description}>
      <Hero title={title} />
      <Navigation sections={navigationSections} />
      <article className="article-reader">
        <h1 className="article-reader__headline">{title}</h1>
        <div>
          <p>
            The page you're trying to access does not exist. Go back to the
            Homepage or find what you're looking for in the menu.
          </p>
          <p>
            Take me back to the <a href="/">Homepage</a> →
          </p>
        </div>
      </article>
    </Layout>
  )
}
```

Reply:
I'll do a follow-up PR adapting the above suggestion. Thank you!